src/Compilers/CSharp/Portable/Binder/Binder_WithExpression.cs

```diff
@@ -69,10 +69,11 @@ private BoundExpression BindWithExpression(WithExpressionSyntax syntax, Diagnost
                         TypeCompareKind.ConsiderEverything,
                         ref useSiteDiagnostics))
                 {
-                    useSiteDiagnostics = null;
                     hasErrors = true;
                     diagnostics.Add(ErrorCode.ERR_NoSingleCloneMethod, syntax.Expression.Location, receiverType);
                 }
+
+                diagnostics.Add(syntax.Expression, useSiteDiagnostics);
```

Comment:
Please add a test that is affected by the change.

Reply:
> Please add a test that is affected by the change.

I do not believe it is worth spending time looking for the scenario. Code like this should be written mechanically without waiting for something to break.

---
In reply to: [450415003](https://github.com/dotnet/roslyn/pull/45664#discussion_r450415003) [](ancestors = 450415003)